src/Api/Controllers/CollectionsController.cs

```diff
@@ -311,6 +307,18 @@ public async Task PostBulkCollectionAccess([FromBody] BulkCollectionAccessReques
             throw new NotFoundException("One or more collections not found.");
         }
 
+        var orgId = collections.First().OrganizationId;
+        if (collections.All(c => c.OrganizationId != orgId))
+        {
+            throw new BadRequestException("All collections must belong to the same organization");
+        }
+
+        // Authorization logic assumes flexible collections is enabled
+        if (!await FlexibleCollectionsIsEnabledAsync(orgId))
```

Comment:
It's because the orgId is taken from the collection objects - which means we need to get the collections from the db, and also check they all belong to the same org (even though this is also covered in the handler).

I rewrote it slightly to use the orgId parameter, it should still have the same effect but addresses your feedback.